### PR TITLE
integrate tx util for faster transactions

### PR DESCRIPTION
**Pull request by alvrs**

### CODE REVIEW

File: packages/client/src/mud/setup.ts [hunk 1]
```diff
@@ -45,11 +47,27 @@ export async function setup() {
...
   }
 
+  if (!signer) throw new Error("No signer");
+  if (!signer.provider) throw new Error("No provider connected to the signer");
+
+  // Create a World contract instance
+  const worldContract = IWorld__factory.connect(
+    networkConfig.worldAddress,
+    signer
+  );
```
Comment (holic) [inline packages/client/src/mud/setup.ts:57]: I did a similar thing in emojimon branch, but opted not to do the signer check, thinking that folks might wanna build a spectator-mode thing?

https://github.com/latticexyz/v2sandbox/pull/17/files#diff-30525edeafc7d169366fa321b43901a7e94518ff244c36942f38618c4446f6bcR56-R59

(that said, I guess we ~always instantiate with a burner wallet and start a drip, so we theoretically should ~always have a signer...)

File: packages/contracts/deploys/31337/latest.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
   "worldAddress": "0xCf7Ed3AccA5a467e9e704C703E8D87F634fB0Fc9",
-  "blockNumber": 19
+  "blockNumber": 9
```
Comment (holic) [inline packages/contracts/deploys/31337/latest.json:3]: we should gitignore this file

File: packages/client/src/index.ts [hunk 3]
```diff
@@ -12,23 +11,44 @@ components.CounterTable.update$.subscribe((update) => {
...
 
-  console.log(txResult);
-  console.log(await txResult.wait());
+  regularTx
+    .then((t) => {
+      console.log("regularTx", t);
+      console.log("regularTx time till sending tx", Date.now() - time, "ms");
+      return t;
+    })
+    .then(async (t) => {
```
Comment (holic) [inline packages/client/src/index.ts:49]: I remember seeing that mixing `.then` with `async` callbacks leads to hard-to-catch promise rejections and generally not advised.

You could write this as `.then((tx) => tx.wait(0)).then((result) => ...);`

File: packages/client/src/index.ts [hunk 4]
```diff
@@ -12,23 +11,44 @@ components.CounterTable.update$.subscribe((update) => {
...
+      const result = await t.wait(0);
```
Comment (holic) [inline packages/client/src/index.ts:50]: Why is this `wait(0)` while above is `wait()`? I thought `wait` defaulted to `1` confirmation, so wouldn't this throw off results?

Comment (alvrs) [reply packages/client/src/index.ts:50]: yeah indeed, gonna this means fastTx is even more faster than regularTx. Gonna test again with this being aligned

File: packages/client/src/mud/setup.ts (hunk 1 above)
Comment (alvrs) [reply packages/client/src/mud/setup.ts:57]: that works! I guess we could add an error in fastTxExecute if no signer is connected?

**Pull request merged**